validator/sawtooth_validator/journal/chain.py

```diff
@@ -473,3 +480,27 @@ def on_block_received(self, block):
         # pylint: disable=broad-except
         except Exception as exc:
             LOGGER.exception(exc)
+
+    def _set_chain_head(self, block):
+        # This is used by journal when it has recieved the genesis block from
+        # the genesis validator
+        if block.previous_block_id == NULL_BLOCK_IDENTIFIER:
+            self._block_store.set_chain_head(block.identifier)
```

Comment:
Also we need to validate the block to build the required state? Not sure what happens if the block publisher tries to build a block on top of a block that the state_root_hash is not present in the merkle tree.